python/ray/rllib/optimizers/multi_gpu.py

```diff
@@ -42,16 +45,16 @@ def _init(self, sgd_batch_size=128, sgd_stepsize=5e-5, num_sgd_iter=10):
         self.load_timer = TimerStat()
         self.grad_timer = TimerStat()
         self.update_weights_timer = TimerStat()
+        self._cur_iter_stats = defaultdict(list)
 
         print("LocalMultiGPUOptimizer devices", self.devices)
         print("LocalMultiGPUOptimizer batch size", self.batch_size)
 
         # List of (feature name, feature placeholder) tuples
         self.loss_inputs = self.local_evaluator.tf_loss_inputs()
-
         # per-GPU graph copies created below must share vars with the policy
-        tf.get_variable_scope().reuse_variables()
```

Comment:
Why remove this?

Reply:
the main issue was just that `TOWER_SCOPE` in `multi_gpu_impl` would introduce a different scope, where variables then could not be reused from the local evaluator. 

Basically, there are three places where the policy graph is built: 
 - the local evaluator
 - the `shared_loss` in the `par_opt`
 - the `device_loss`es in the `par_opt`

`tf.get_variable_scope().reuse_variables()` gets the scope where the local_evaluator copy resides, but the `TOWER_SCOPE_NAME` gates off the `par_opt` copies. 

On a separate branch right now, I tried making the local evaluator and `shared_loss` the same copy, but it doesn't work yet and I think I need an extra sync between the `device_losses` graph and the local evaluator graph...